ionoscloud.go

```diff
@@ -865,33 +839,74 @@ func (d *Driver) Create() (err error) {
 		ipsForAttachedNic = reservedIps // For public NICs we use the generated IPs
 	}
 
+	attachedNics := sdkgo.NewNicsWithDefaults()
+
 	lanId, _ := strconv.Atoi(d.LanId)
-	nic, err := d.client().CreateAttachNIC(d.DatacenterId, d.ServerId, d.MachineName, d.NicDhcp, int32(lanId), ipsForAttachedNic)
+	lanId_int32 := int32(lanId)
+	nicProperties := &sdkgo.NicProperties{
+		Name: &d.MachineName,
+		Lan:  &lanId_int32,
+		Ips:  ipsForAttachedNic,
+		Dhcp: &d.NicDhcp,
+	}
+
+	attachedNics.Items = &[]sdkgo.Nic{
+		{
+			Properties: nicProperties,
+		},
+	}
+
+	for _, additionalLanId := range d.AdditionalLansIds {
+		dhcp_true := true
+		additionalLanId_int32 := int32(additionalLanId)
+		additionalNicName := d.MachineName + " " + fmt.Sprint(additionalLanId)
+		nicProperties = &sdkgo.NicProperties{
+			Name: &additionalNicName,
+			Lan:  &additionalLanId_int32,
+			Ips:  nil,
+			Dhcp: &dhcp_true,
+		}
+		additionalNic := sdkgo.Nic{
+			Properties: nicProperties,
+		}
+		*attachedNics.Items = append(*attachedNics.Items, additionalNic)
+	}
+
+	serverToCreate.Entities.SetNics(*attachedNics)
+
+	server, err := d.client().CreateServer(d.DatacenterId, serverToCreate)
 	if err != nil {
-		// TODO: Duplicated
+		// TODO: Export to a func
 		log.Warn(rollingBackNotice)
 		if removeErr := d.Remove(); removeErr != nil {
-			return fmt.Errorf("failed to create machine due to error: %w\n Removing created resources: %v", fmt.Errorf("error attaching NIC: %w", err), removeErr)
+			return fmt.Errorf("failed to create server due to error: %w\n Removing created resources: %v", err, removeErr)
 		}
 		return err
 	}
-	for _, additionalLanId := range d.AdditionalLansIds {
-		additionalNic, err := d.client().CreateAttachNIC(d.DatacenterId, d.ServerId, d.MachineName, true, int32(additionalLanId), nil)
-		if err != nil {
-			// TODO: Duplicated
-			log.Warn(rollingBackNotice)
-			if removeErr := d.Remove(); removeErr != nil {
-				return fmt.Errorf("failed to create machine due to error: %w\n Removing created resources: %v", fmt.Errorf("error attaching additional NIC: %w", err), removeErr)
-			}
-			return err
-		}
-		if nicId, ok := additionalNic.GetIdOk(); ok && nicId != nil {
-			d.AdditionalNicsIds = append(d.AdditionalNicsIds, *nicId)
-		}
+	if serverId, ok := server.GetIdOk(); ok && serverId != nil {
+		d.ServerId = *serverId
+		log.Debugf("Server ID: %v", d.ServerId)
+	}
+
+	server, err = d.client().GetServer(d.DatacenterId, d.ServerId, 2)
+	if err != nil {
+		return fmt.Errorf("error getting server by id: %w", err)
 	}
-	if nicId, ok := nic.GetIdOk(); ok && nicId != nil {
-		d.NicId = *nic.Id
-		log.Debugf("Nic ID: %v", d.NicId)
+	volumes, ok := server.Entities.GetVolumesOk()
+	if !ok {
+		return fmt.Errorf("error getting server: d.ServerId is empty")
```

Comment:
I think this error message is wrong, wouldn't this print `error getting server: d.ServerId is empty` when in fact there are no volumes on that server?